### PR TITLE
Allow ability to NOT prefix metric including app flavor. This shhould…

### PR DESCRIPTION
**Pull request by mikebell90**
… be judiciously done, but gc depends on it, sigh.

### CODE REVIEW

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java [hunk 1]
```diff
@@ -71,6 +71,9 @@
     @Value("${ot.graphite.reporting-period:PT10s}")
     private Duration reportingPeriod;
 
+    @Value("${ot.graphite.reporting.include.flavors:#{true}}")
```
Comment (mikebell90) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java]: Default to old behavior

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java [hunk 2]
```diff
@@ -138,14 +141,14 @@ void close() {
...
-    static String getPrefix(ServiceInfo serviceInfo, AppInfo appInfo) {
+    static String getPrefix(ServiceInfo serviceInfo, AppInfo appInfo, boolean includeFlavorInPrefix ) {
         final String applicationName = serviceInfo.getName();
         final EnvInfo env = appInfo.getEnvInfo();
         final Integer i = appInfo.getInstanceNumber();
         if (env.getType() == null || env.getLocation() == null || i == null) {
             return null;
         }
-        final String name = env.getFlavor() == null ? applicationName : applicationName + "-" + env.getFlavor();
+        final String name = env.getFlavor() == null || (!includeFlavorInPrefix) ? applicationName : applicationName + "-" + env.getFlavor();
```
Comment (mikebell90) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java]: Condition.

File: otj-metrics-core/src/test/java/com/opentable/metrics/graphite/GraphiteReporterPrefixTest.java [hunk 3]
```diff
@@ -41,31 +40,55 @@
     private ScheduledReporter reporter;
 
     @Test
-    public void withFlavor() {
-        final String prefix = prefixFrom("type-location.flavor", "0");
+    public void withFlavorEnabled() {
+        final String prefix = prefixFrom("type-location.flavor", "0", true);
```
Comment (mikebell90) [inline otj-metrics-core/src/test/java/com/opentable/metrics/graphite/GraphiteReporterPrefixTest.java]: New tests.

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java (hunk 2 above)
Comment (adamzr) [reply otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java]: I think this should use if blocks.

**Comment by mikebell90** [comment]
@aquariusrick @dkaukov says there's a better way? Please discuss.

**Comment by aquariusrick** [comment]
This change is for `umami-guest-share` application which is using flavor.  Previously it was using `metrics-spring` library.  Now when upgrading to this, it's adding `-flavor` to our metrics, and we don't want to change our metrics.  The main reason I don't want it to change is because our old metrics will be purged after 30 days if we change the prefix.

**Comment by aquariusrick** [comment]
> @aquariusrick @dkaukov says there's a better way? Please discuss.

I understood that @dkaukov suggested to update the dashboard.  I'm trying to avoid this, not just for the sake of not wanting to, but also not wanting lose historical metrics.

**Pull request merged**